notebook/nbextensions.py

```diff
@@ -299,118 +355,300 @@ def _config_file_name_default(self):
     def install_extensions(self):
         if len(self.extra_args)>1:
             raise ValueError("only one nbextension allowed at a time.  Call multiple times to install multiple extensions.")
-        install_nbextension(self.extra_args[0],
-            overwrite=self.overwrite,
-            symlink=self.symlink,
-            verbose=self.verbose,
-            user=self.user,
-            prefix=self.prefix,
-            destination=self.destination,
-            nbextensions_dir=self.nbextensions_dir,
+        if self.python:
+            install_nbextension_python(self.extra_args[0],
+                overwrite=self.overwrite,
+                symlink=self.symlink,
+                verbose=self.verbose,
+                user=self.user,
+                sys_prefix=self.sys_prefix,
+                prefix=self.prefix,
+                nbextensions_dir=self.nbextensions_dir
+            )
+        else:
+            install_nbextension(self.extra_args[0],
+                overwrite=self.overwrite,
+                symlink=self.symlink,
+                verbose=self.verbose,
+                user=self.user,
+                prefix=self.prefix,
+                destination=self.destination,
+                nbextensions_dir=self.nbextensions_dir,
         )
     
     def start(self):
         if not self.extra_args:
-            for nbext in jupyter_path('nbextensions'):
-                if os.path.exists(nbext):
-                    print("Notebook extensions in %s:" % nbext)
-                    for ext in os.listdir(nbext):
-                        print(u"    %s" % ext)
+            self.log.warn('Please specify an nbextension to install')
         else:
             try:
                 self.install_extensions()
             except ArgumentConflict as e:
-                print(str(e), file=sys.stderr)
+                self.log.info(str(e), file=sys.stderr)
                 self.exit(1)
 
 
-class EnableNBExtensionApp(JupyterApp):
-    name = "jupyter nbextension enable"
+_toggle_flags = {
```

Comment:
I imagine these flags should be set on BaseNBExtensionApp, not Toggle, since they are common, right?